src/components/AngerNoodle/helperFunctions/gameLogic.js

```diff
@@ -0,0 +1,110 @@
+import { makeRandomPositionIndex, copyBoard, copySnake, angryMessages } from "./data";
+
+const computerNextFrame = (shouldDeleteTail, updateShouldDeleteTail, snake, updateSnake, gridSize, board, updateBoard, direction, wallsAreLava, treatCoords, updateTreatCoords, currentScore, updateScore, highScore, updateHighScore, gameStateRef, snakeRef, updateAngryMessage) => {
+  let score = currentScore;
+  const currentHighScore = highScore;
+  let newTreatCoords = null;
+  const gameState = gameStateRef;
+  const currentSnakeRef = snakeRef;
+  const boardCopy = copyBoard(board);
+  const snakeCopy = copySnake(snake);
+  const head = snakeCopy[snake.length - 1];
+  const boundary = gridSize - 1;
+
+  const removeTail = () => {
+    snakeCopy.splice(0, 1);
```

Comment:
even though snakeCopy is scoped to this function, I'd recommend having it be a parameter of it to be safe. It'll make it much easier to toss around in case you ever decide that removeTail (or any of the other functions below) needs to exist in another part of your application / if you ever refactor variables that they depend on